src/Umbraco.Core/Trees/MenuItemList.cs

```diff
@@ -44,7 +47,7 @@ public MenuItem Add<T>(ILocalizedTextService textService, bool hasSeparator = fa
         private MenuItem CreateMenuItem<T>(ILocalizedTextService textService, bool hasSeparator = false, bool opensDialog = false)
             where T : IAction
         {
-            var item = Current.Actions.GetAction<T>();
+            var item = _actionCollection.GetAction<T>();
```

Comment:
I'm assuming this is the underlying reason why we need the MenuItemCollectionFactory?